Leanplum-SDK/Classes/LPActionManager.m

```diff
@@ -77,26 +70,24 @@ - (void)leanplum_application:(UIApplication *)application didRegisterForRemoteNo
         [self leanplum_disableAskToAsk];
     }
     
-    // Send push token on every launch. This is required because push token may not be sent if the
-    // user force quit the app. This is due to how LPRequestStorage keeps the data in memory and
-    // we don't support response block on batch.
+    // Format push token.
     NSString *formattedToken = [deviceToken description];
     formattedToken = [[[formattedToken stringByReplacingOccurrencesOfString:@"<" withString:@""]
                        stringByReplacingOccurrencesOfString:@">" withString:@""]
                       stringByReplacingOccurrencesOfString:@" " withString:@""];
-    [Leanplum onStartResponse:^(BOOL success) {
-        LP_END_USER_CODE
-        [[LeanplumRequest post:LP_METHOD_SET_DEVICE_ATTRIBUTES
-                        params:@{LP_PARAM_DEVICE_PUSH_TOKEN: formattedToken}] send];
-        LP_BEGIN_USER_CODE
-    }];
 
-    // Update push token if it's different.
-    NSString *tokenKey = [self leanplum_createTokenKey];
+    // Send push token if we don't have one and when the token changed.
+    // We no longer send in start's response because saved push token will be send in start too.
+    NSString *tokenKey = [Leanplum pushTokenKey];
     NSString *existingToken = [[NSUserDefaults standardUserDefaults] stringForKey:tokenKey];
-    if (![existingToken isEqualToString: formattedToken]) {
-        [[NSUserDefaults standardUserDefaults] setObject: formattedToken forKey:tokenKey];
+    if (!existingToken || ![existingToken isEqualToString:formattedToken]) {
+        [[NSUserDefaults standardUserDefaults] setObject:formattedToken forKey:tokenKey];
         [[NSUserDefaults standardUserDefaults] synchronize];
+        
+        if ([LPInternalState sharedState].calledStart) {
```

Comment:
You're right. calledStart will always be true. I was trying to avoid sending request if the token was sent with start but I was over optimizing it. Thanks for the catch.